tracer/src/Datadog.Trace/ClrProfiler/AutoInstrumentation/Grpc/GrpcLegacy/Client/GrpcLegacyClientCommon.cs

```diff
@@ -242,20 +242,13 @@ private static Span CreateInactiveSpan(Tracer tracer, string? methodFullName)
             var operationName = tracer.CurrentTraceSettings.Schema.Client.GetOperationNameForProtocol("grpc");
             var serviceName = tracer.CurrentTraceSettings.Schema.Client.GetServiceName(component: "grpc-client");
             var tags = tracer.CurrentTraceSettings.Schema.Client.CreateGrpcClientTags();
-            var span = tracer.StartSpan(operationName, tags, serviceName: serviceName, addToTraceContext: false);
             tags.SetAnalyticsSampleRate(IntegrationId.Grpc, tracer.Settings, enabledWithGlobalSetting: false);
             tracer.CurrentTraceSettings.Schema.RemapPeerService(tags);
 
+            var span = tracer.StartSpan(operationName, tags, serviceName: serviceName, addToTraceContext: false);
             span.Type = SpanTypes.Grpc;
             span.ResourceName = methodFullName;
 
-            if (span.Context.TraceContext.SamplingPriority == null)
```

Comment:
Just for my understanding, we don't need to do this explicitly here, because the propagator does it implicitly right?

Reply:
Right. Before this PR, we triggered a sampling decision when the first span was added to the trace in `TraceContext.AddSpan(Span)`. Since this `span` is _not_ added to the trace (see `tracer.StartSpan(..., addToTraceContext: false)` a few lines above), we were triggering the sampling decision manually.

With this PR, we no longer trigger the sampling decision when a span is added to the trace. Instead we delay as long as possible and make the sampling decision only when we need to propagate it, either to the trace agent (as span tags) or to a downstream service (as headers).